### PR TITLE
Removes integration tests

### PR DESCRIPTION
**Pull request by chrislloyd**
We now expect that integration tests lie closer to the consuming
libraries. They're still _really_ important, however a lot of these
tests rely on the environment in which they're run. In particular,
Masonry's tests are highly dependent on the actual things you put
inside Masonry.

The tests themselves still exist internally within Pinterest so no
coverage is actually being lost. The upside is that we can
significantly simply the project setup here.

In addition I've:

* Updated Travis to also test against lts/dubnium seeing as
   `yarn install` was throwing errors with it locally
* Updated CRA and a couple of other deps to make sure
   everything works and installs correctly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,12 +13,10 @@
   },
   "devDependencies": {
     "babel-core": "^6.26.0",
-    "babel-eslint": "^10.0.1",
+    "babel-eslint": "9.0.0",
```
Comment (btraut) [inline package.json]: You downgraded the version of eslint? Was that intentional?

File: package.json [hunk 2]
```diff
@@ -14,12 +13,10 @@
...
     "babel-jest": "^23.4.0",
```
Comment (btraut) [inline package.json]: Are there any jest files left? I suspect there's more deps we can remove.

File: package.json (hunk 1 above)
Comment (chrislloyd) [reply package.json]: Yep - `create-react-app@2` now requires exact versions of eslint. Not awesome but also not terrible (not sure what that version got us). For this and a few other reasons I don't actually think CRA is a great fit anymore for the docs so will address in further PRs.

File: package.json (hunk 2 above)
Comment (chrislloyd) [reply package.json]: There are tonnes of jest files :) Lots of unit tests and snapshots for components (which are great IMHO).

**Pull request merged**